demonstrations/tutorial_noisy_circuits.py

```diff
@@ -0,0 +1,228 @@
+# are described as vectors, a different method is needed for simulating mixed-state quantum
+# computations. This is the purpose of PennyLane's
+#  device, which provides native
+# support for mixed states and for simulating all types of noise.
+#
+# Let's use ``default.mixed`` to simulate a simple circuit for preparing the
+# Bell state :math:`|\psi\rangle=\frac{1}{\sqrt{2}}(|00\rangle+|11\rangle)`. We ask the QNode to
+# return the expectation value of :math:`Z_0\otimes Z_1`:
+#
+import pennylane as qml
+from pennylane import numpy as np
+
+qml.enable_tape()
+dev = qml.device('default.mixed', wires=2)
+ZZ = qml.PauliZ(0)@qml.PauliZ(1)
+
+
+@qml.qnode(dev)
+def circuit():
+    qml.Hadamard(wires=0)
+    qml.CNOT(wires=[0, 1])
+    return qml.expval(ZZ)
+
+
+print(f"QNode output = {circuit():.4f}")
+
+######################################################################
+# The device stores the output state as a density matrix. In this case, since we haven't
+# introduced any noise, the density matrix is equal to the rank-one
+# projector :math:`|\psi\rangle\langle\psi|, where :math:`|\psi\rangle=\frac{1}{\sqrt{
+# 2}}(|00\rangle + |11\rangle)`.
+
+print(np.real(dev.state))
+
+######################################################################
+# Just as mixed states are represented by density matrices, incoherent noise is modelled by
+# *quantum channels*, which are linear, completely positive, and trace-preserving (`CPTP
+# <https://www.quantiki.org/wiki/channel-cp-map>`__) maps. A convenient strategy for representing
+# quantum channels is to employ Kraus operators :math:`K = \{K_i\}` satisfying the condition
+# :math:`\sum_i K_{i}^{\dagger} K_i = \mathcal{I}`. From an initial state :math:`\rho`, the output
+# state after the action of a channel :math:`E` is:
+#
+# .. math::
+#
+#     \rho' = E(\rho) = \sum_i K_i \rho K_{i}^{\dagger}.
+#
+# The action of a quantum channel can thus be interpreted as applying the transformation
+# :math:`K_i` with probability :math:`p_i = \text{Tr}[K_i \rho K_{i}^{\dagger}]`. For
+# example, consider the bit flip channel. It describes a transformation that flips the state of
+# a qubit (applies an X gate) with probability :math:`p`. Its Kraus operators are
+#
+# .. math::
+#
+#     K_0 = (1-p)\begin{pmatrix}1 & 0\\ 0 & 1\end{pmatrix}
+#     K_1 = p\begin{pmatrix}0 & 1\\ 1 & 0\end{pmatrix}.
+#
+# This channel can be implemented in PennyLane using the :func:`~.pennylane.BitFlip` operation.
+# Let's see what happens when we simulate this type of noise acting on
+# both qubits in the circuit. We'll evaluate the QNode for different bit flip probabilitites.
+#
+
+
+@qml.qnode(dev)
+def bitflip_circuit(p):
+    qml.Hadamard(wires=0)
+    qml.CNOT(wires=[0, 1])
+    qml.BitFlip(p, wires=0)
+    qml.BitFlip(p, wires=1)
+    return qml.expval(ZZ)
+
+
+ps = [0.001, 0.01, 0.1, 0.2]
+for p in ps:
+    print(f"QNode output = {bitflip_circuit(p):.4f}")
+
+print(f"Output state = {np.real(dev.state)}")
```

Comment:
```suggestion
print(f"Output state = \n{np.real(dev.state)}")
```

(Helps with readability)

Reply:
I didn't know about `\n`! Thanks for the suggestion and lesson!